packages/contracts-periphery/contracts/universal/drippie/Drippie.sol

```diff
@@ -0,0 +1,245 @@
+        require(
+            drips[_name].status != _status,
+            "Drippie: cannot set drip status to same status as before"
+        );
+
+        // If the user is trying to archive this drip, make sure the drip has been paused. We do
+        // not allow users to archive active drips so that the effects of this action are more
+        // abundantly clear.
+        if (_status == DripStatus.ARCHIVED) {
+            require(
+                drips[_name].status == DripStatus.PAUSED,
+                "Drippie: drip must be paused to be archived"
+            );
+        }
+
+        // If we made it here then we can safely update the status.
+        drips[_name].status = _status;
+        emit DripStatusUpdated(_name, drips[_name].status);
+    }
+
+    /**
+     * Triggers a drip. This function is deliberately left as a public function because the
+     * assumption being made here is that setting the drip to ACTIVE is an affirmative signal that
+     * the drip should be executable according to the drip parameters, drip check, and drip
+     * interval. Note that drip parameters are read entirely from the state and are not supplied as
+     * user input, so there should not be any way for a non-authorized user to influence the
+     * behavior of the drip.
+     *
+     * @param _name Name of the drip to trigger.
+     */
+    function drip(string memory _name) external {
+        DripState storage state = drips[_name];
+
+        // Only allow active drips to be executed, an obvious security measure.
+        require(
+            state.status == DripStatus.ACTIVE,
+            "Drippie: selected drip does not exist or is not currently active"
+        );
+
+        // Don't drip if the drip interval has not yet elapsed since the last time we dripped. This
+        // is a safety measure that prevents a malicious recipient from, e.g., spending all of
+        // their funds and repeatedly requesting new drips. Limits the potential impact of a
+        // compromised recipient to just a single drip interval, after which the drip can be paused
+        // by the owner address.
+        require(
+            state.last + state.config.interval <= block.timestamp,
+            "Drippie: drip interval has not elapsed since last drip"
+        );
+
+        // Make sure we're allowed to execute this drip.
+        require(
+            state.config.dripcheck.check(state.config.checkparams),
+            "Drippie: dripcheck failed so drip is not yet ready to be triggered"
+        );
+
+        // Update the last execution time for this drip before the call. Note that it's entirely
+        // possible for a drip to be executed multiple times per block or even multiple times
+        // within the same transaction (via re-entrancy) if the drip interval is set to zero. Users
+        // should set a drip interval of 1 if they'd like the drip to be executed only once per
+        // block (since this will then prevent re-entrancy).
```

Comment:
I can't imagine a case where this would actually be desirable, and in the case the the sequencer is being drained we'd want to ensure that it can't easily be refilled. 
I'd suggest either: 
1. disallow `_config.interval==0` in the contracts
2. warn or throw an error in the ts when attempting to create a drip with 0 interval.

Reply:
I'd like to leave the functionality in there. I'll make sure that there's a very obvious warning in the typescript tooling.